cpp/src/arrow/python/io.cc

```diff
@@ -216,5 +216,19 @@ Status PyOutputStream::Write(const void* data, int64_t nbytes) {
   return file_->Write(data, nbytes);
 }
 
+// ----------------------------------------------------------------------
+// Foreign buffer
+
+Status PyForeignBuffer::Make(const uint8_t* data, int64_t size, PyObject* base,
```

Comment:
Can you add the description of the JIRA ticket somewhere in the code? I was not aware what went wrong in the initial implementation only from reading the change but I needed the context of the JIRA issue.

Reply:
Ok, I added an explanation.